checks/shell_download_validate.go

```diff
@@ -401,7 +400,63 @@ func isPipUnpinnedDownload(cmd []string) bool {
 		}
 	}
 
-	return isInstalled
+	return isInstall
+}
+
+func isPythonCommand(cmd []string) bool {
+	for _, pi := range pythonInterpreters {
+		if isBinaryName(pi, cmd[0]) {
+			return true
+		}
+	}
+	return false
+}
+
+func isPythonPipInstall(cmd []string) bool {
+	if !isPythonCommand(cmd) {
```

Comment:
can also rewrite this to share more logic between `python -m pip` and `pip`.

e.g. something like:

```go
func isPythonPipInstall(cmd []string) bool {
  pipCommand, ok := extractPipCommand(cmd)
  if !ok {
    return false
  }

  return isPipInstall(pipCommand) 
}
```

Reply:
great suggestion. Done